### PR TITLE
chore: Add Sidebar with Pages Tab

### LINKED ISSUE #27936: [Feature]: Add app sidebar with Pages pane state
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

https://www.figma.com/proto/GTOUFv2RYxDjmxfbGYoWpn/IA-changes?type=design&node-id=408-40260&t=tjYviuVkHVLgh2NQ-0&scaling=min-zoom&page-id=408%3A23992

### Why should this be worked on?

Part of IA redesign

### PR DESCRIPTION
**Pull request by hetunandu**
## Description
Adds the sidebar with just the pages tab for now. This is just a visual change, with no way to switch to any other state on the app sidebar. This change is behind a feature flag.

#### PR fixes following issue(s)
Fixes #27936

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by KelvinOm — DISMISSED** [review]

File: app/client/src/components/Sidebar/SidebarButton.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+const Container = styled.div`
+  display: flex;
+  justify-content: center;
+  flex-direction: column;
+  height: 64px;
+  width: 50px;
+  svg {
+    width: 24px;
+    height: 24px;
```
Comment (akash-codemonk) [inline app/client/src/components/Sidebar/SidebarButton.tsx:19]: Icon size lg is 24px

**Review by akash-codemonk — DISMISSED** [review]

**Comment by akash-codemonk** [comment]
dropdown

There is something wrong with the dropdowns in the new ds version. cc @albinAppsmith

**Review by akash-codemonk — APPROVED** [review]

**Pull request merged**